In [1]:
for i in range(1, 5):
    for j in range(1, 5):
        for k in range(1, 5):
            if i!=j and i!=k and j!=k:
                print(i,j,k)

1 2 3
1 2 4
1 3 2
1 3 4
1 4 2
1 4 3
2 1 3
2 1 4
2 3 1
2 3 4
2 4 1
2 4 3
3 1 2
3 1 4
3 2 1
3 2 4
3 4 1
3 4 2
4 1 2
4 1 3
4 2 1
4 2 3
4 3 1
4 3 2


In [2]:
i = int(input("输入利润"))
arr = [1000000, 600000, 400000, 200000, 100000, 0]
rat = [0.01, 0.015, 0.03, 0.05, 0.075, 0.1]
r = 0
for idx in range(0, 6):
    if i > arr[idx]:
        r+=(i-arr[idx])*rat[idx]
        print((i-arr[idx])*rat[idx])
        i=arr[idx]
print(r)

输入利润1222
122.2
122.2


In [3]:
for x in range(1, 13):
    a = 84/x - x/2
    if int(a)==a:
        n=a**2-100
        print(n)

1581.0
261.0
21.0
-99.0


 ###  输入某年某月某日，判断这一天是这一年的第几天

In [4]:
year = int(input("输入年份"))
month = int(input("输入月份"))
day = int(input("输入日"))
def calculateDay(year, month, day):
    days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31, 30]
    counterResult = sum(days[0:month-1]) + day
    if (year%4==0) or ((year%400==0) and (year%100!=0)):
        if month>2:
            counterResult+=1
    return counterResult
calculateDay(year, month, day)

输入年份2332
输入月份3
输入日4


64

### 输入三个整数x,y,z，请把这三个数由小到大输出

In [5]:
def printNumberByOrder():
    arr = []
    for i in range(0, 3):
        arr.append(int(input("输入第"+str(i+1)+"个整数")))
    arr.sort()
    print(arr)
printNumberByOrder()

输入第1个整数1
输入第2个整数23
输入第3个整数6
[1, 6, 23]


### 斐波那契数列 

In [6]:
def fib(n):
    if n==1 or n==2:
        return 1
    return fib(n-1)+fib(n-2)
print(fib(10))

55


### 循环斐波那契数列 

In [7]:
def loopFib(n):
    arr = [1, 1]
    for i in range(2, n):
        arr.append(arr[i-1]+arr[i-2])
    print(arr)
loopFib(100)

[1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578, 5702887, 9227465, 14930352, 24157817, 39088169, 63245986, 102334155, 165580141, 267914296, 433494437, 701408733, 1134903170, 1836311903, 2971215073, 4807526976, 7778742049, 12586269025, 20365011074, 32951280099, 53316291173, 86267571272, 139583862445, 225851433717, 365435296162, 591286729879, 956722026041, 1548008755920, 2504730781961, 4052739537881, 6557470319842, 10610209857723, 17167680177565, 27777890035288, 44945570212853, 72723460248141, 117669030460994, 190392490709135, 308061521170129, 498454011879264, 806515533049393, 1304969544928657, 2111485077978050, 3416454622906707, 5527939700884757, 8944394323791464, 14472334024676221, 23416728348467685, 37889062373143906, 61305790721611591, 99194853094755497, 160500643816367088, 259695496911122585, 420196140727489673, 679891637638612258, 110008777836

### 拷贝列表 

In [8]:
a = [1, 2, 3]
b = a.copy()
c = a[:]
print(b)
print(c)

[1, 2, 3]
[1, 2, 3]


### 输出乘法口诀表

In [9]:
def printPlusTable(n):
    for i in range(1, n+1):
        counter = 1
        stack = []
        while counter<=i:
            stack.append(str(i)+"*"+str(counter)+"="+str(i*counter)+" ")
            counter+=1
        print(stack)
printPlusTable(9)

['1*1=1 ']
['2*1=2 ', '2*2=4 ']
['3*1=3 ', '3*2=6 ', '3*3=9 ']
['4*1=4 ', '4*2=8 ', '4*3=12 ', '4*4=16 ']
['5*1=5 ', '5*2=10 ', '5*3=15 ', '5*4=20 ', '5*5=25 ']
['6*1=6 ', '6*2=12 ', '6*3=18 ', '6*4=24 ', '6*5=30 ', '6*6=36 ']
['7*1=7 ', '7*2=14 ', '7*3=21 ', '7*4=28 ', '7*5=35 ', '7*6=42 ', '7*7=49 ']
['8*1=8 ', '8*2=16 ', '8*3=24 ', '8*4=32 ', '8*5=40 ', '8*6=48 ', '8*7=56 ', '8*8=64 ']
['9*1=9 ', '9*2=18 ', '9*3=27 ', '9*4=36 ', '9*5=45 ', '9*6=54 ', '9*7=63 ', '9*8=72 ', '9*9=81 ']


### 暂停一秒的输出

In [10]:
import time
myD = {1:'a', 2:'b'}
for key, value in dict.items(myD):
    time.sleep(2)
    print(key, value)

1 a
2 b


### 输入格式化时间

In [11]:
import time
print(time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time())))

2018-03-19 10:53:39


### 判断101-200之间有多少个素数，并输出所有素数

In [12]:
from math import sqrt
def findPrimeNumber():
    arr = []
    for i in range(101, 201):
        flag = True
        k = int(sqrt(i+1))+1
        for j in range(2, k):
            if i%j==0:
                flag = False
                break
        if flag:
            arr.append(i)
    print(arr)
findPrimeNumber()

[101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, 199]


### 打印100到999水仙花数

In [13]:
def printSxhNumber():
    arr = []
    for i in range(100, 1000):
        s = str(i)
        if (int(s[0])**3+int(s[1])**3+int(s[2])**3)==i:
            arr.append(i)
    print(arr)
printSxhNumber()

[153, 370, 371, 407]


### 分解整数质因数

In [14]:
def decouple(n):
    arr = []
    while n!=1:
        for i in range(2, n+1):
            if n%i==0:
                arr.append(i)
                n=int(n/i)
                break;
    return arr
decouple(323)

[17, 19]

### 学习成绩>=90分的同学用A表示，60-89分之间的用B表示，60分以下的用C表示。

In [15]:
def defineScore(n):
    scoretype = ''
    if n >= 90:
        scoretype = 'A'
    elif n >= 60:
        scoretype = 'B'
    else:
        scoretype = 'C'
    print('%d 属于 %s' % (n, scoretype))
defineScore(90)

90 属于 A


### 输出指定格式的日期

In [16]:
import datetime
# print date format
print(datetime.date.today().strftime('%d/%m/%Y'))
mydate = datetime.date(1949, 10, 1)
print(mydate.strftime('%d/%m/%Y'))
# date calculate
mydateNextDate = mydate+datetime.timedelta(days=1)
print(mydateNextDate.strftime('%d/%m/%Y'))
# date replace
mydatereplaceyear = mydate.replace(year=mydate.year+1)
print(mydatereplaceyear.strftime('%d/%m/%Y'))

19/03/2018
01/10/1949
02/10/1949
01/10/1950


### 输入一行字符，分别统计出其中英文字母、空格、数字和其它字符的个数。

In [17]:
import string
s = input('input a string:\n')
letters = 0
space = 0
digit = 0
others = 0
for c in s:
    if c.isalpha():
        letters+=1
    elif c.isspace():
        space+=1
    elif c.isdigit():
        digit+=1
    else:
        others+=1
print('char = %d,space = %d,digit = %d,others = %d' % (letters,space,digit,others))

input a string:
222
char = 0,space = 0,digit = 3,others = 0


### 求s=a+aa+aaa+aaaa+aa...a的值，其中a是一个数字。例如2+22+222+2222+22222(此时共有5个数相加)，几个数相加由键盘控制

In [18]:
def strangeCal(n, a):
    sum = 0
    for i in range(1, n+1):
        sum+=a*(10**(i-1))*(n-i+1)
    print(sum)
strangeCal(5,2)

24690


### 一个数如果恰好等于它的因子之和，这个数就称为"完数"。例如6=1＋2＋3.编程找出1000以内的所有完数

In [19]:
def decoupleNumber(n):
    arr = []
    for i in range(1, int(n/2)+1):
        if n%i==0:
            arr.append(i)
    return arr
def findCompleteNumber():
    comNum = []
    for i in range(1, 1000):
        arr = decoupleNumber(i)
        if sum(arr)==i:
            comNum.append(i)
    return comNum
print(findCompleteNumber())

[6, 28, 496]


### 一球从100米高度自由落下，每次落地后反跳回原高度的一半；再落下，求它在第10次落地时，共经过多少米？第10次反弹多高？

In [20]:
def calcBounceBack(n):
    length = 0
    hight = 100
    for i in range(1, n+1):
        length+=hight
        hight/=2
        if(i!=n):
            length+=hight
    print('第 %d 次落地经过 %f 米， 第%d次反弹 %f 米高' % (n, length, n, hight))
calcBounceBack(10)

第 10 次落地经过 299.609375 米， 第10次反弹 0.097656 米高


### 猴子吃桃问题：猴子第一天摘下若干个桃子，当即吃了一半，还不瘾，又多吃了一个第二天早上又将剩下的桃子吃掉一半，又多吃了一个。以后每天早上都吃了前一天剩下的一半零一个。到第10天早上想再吃时，见只剩下一个桃子了。求第一天共摘了多少。

In [21]:
def monkeyPeach():
    day = 10
    peach = 1
    while day > 1:
        day-=1
        peach+=1
        peach*=2
        print('day %d has %d peaches' % (day, peach))
monkeyPeach()

day 9 has 4 peaches
day 8 has 10 peaches
day 7 has 22 peaches
day 6 has 46 peaches
day 5 has 94 peaches
day 4 has 190 peaches
day 3 has 382 peaches
day 2 has 766 peaches
day 1 has 1534 peaches


### 两个乒乓球队进行比赛，各出三人。甲队为a,b,c三人，乙队为x,y,z三人。已抽签决定比赛名单。有人向队员打听比赛的名单。a说他不和x比，c说他不和x,z比，请编程序找出三队赛手的名单。

### 打印出如下图案（菱形）:

### 有一分数序列：2/1，3/2，5/3，8/5，13/8，21/13...求出这个数列的前20项之和

In [22]:
import fractions as fa
def calculateSub():
    a, b= 1, 2
    temp = 0
    sumResult = 0
    arr = []
    for i in range(1, 21):
        arr.append(fa.Fraction(b,a));
        temp = a+ b
        a=b
        b=temp
    return arr
arr = calculateSub()
print(arr)
print(float(sum(arr)))

[Fraction(2, 1), Fraction(3, 2), Fraction(5, 3), Fraction(8, 5), Fraction(13, 8), Fraction(21, 13), Fraction(34, 21), Fraction(55, 34), Fraction(89, 55), Fraction(144, 89), Fraction(233, 144), Fraction(377, 233), Fraction(610, 377), Fraction(987, 610), Fraction(1597, 987), Fraction(2584, 1597), Fraction(4181, 2584), Fraction(6765, 4181), Fraction(10946, 6765), Fraction(17711, 10946)]
32.660260798641644


### 求1+2!+3!+...+20!的和。

In [23]:
n = 0
s = 0
t = 1
for n in range(1,21):
    t *= n
    print(t)
    s += t
print('1! + 2! + 3! + ... + 20! = %d' % s)

1
2
6
24
120
720
5040
40320
362880
3628800
39916800
479001600
6227020800
87178291200
1307674368000
20922789888000
355687428096000
6402373705728000
121645100408832000
2432902008176640000
1! + 2! + 3! + ... + 20! = 2561327494111820313


### 利用递归方法求5!。

In [24]:
def loopFun(n):
    if n ==1:
        return 1
    else:
        return loopFun(n-1)*n
print(loopFun(5))

120


### 利用递归函数调用方式，将所输入的5个字符，以相反顺序打印出来。

In [25]:
def printReverString(s):
    if len(s)==1:
        return s
    else:
        return s[len(s)-1] + printReverString(s[:len(s)-1])
printReverString('abcde')

'edcba'

### 有5个人坐在一起，问第五个人多少岁？他说比第4个人大2岁。问第4个人岁数，他说比第3个人大2岁。问第三个人，又说比第2人大两岁。问第2个人，说比第一个人大两岁。最后问第一个人，他说是10岁。请问第五个人多大？

In [26]:
def askAge(n):
    if n==1:
        return 10
    else:
        return askAge(n-1)+2
print(askAge(5))

18


### 给一个不多于5位的正整数，要求：一、求它是几位数，二、逆序打印出各位数字。

In [27]:
def printNumberRevers(n):
    arr = []
    for i in range(0, len(str(n))):
        arr.append(str(n)[i])
    print("%d 位数是 %d, 逆序输出是： " % (n, len(arr)))
    arr.reverse()
    print(arr)
printNumberRevers(12345)

12345 位数是 5, 逆序输出是： 
['5', '4', '3', '2', '1']


In [28]:
n =12344
print(len(str(n)))

5


### 一个5位数，判断它是不是回文数。即12321是回文数，个位与万位相同，十位与千位相同。

In [29]:
def isLoopNumber(n):
    a = str(n)[::-1]
    return int(a)==n
isLoopNumber(12321)

True

### 按相反的顺序输出列表的值。

In [30]:
a = ["one", "two", "three"]
for i in a[::-1]:
    print(i)

three
two
one


### 按逗号连接分隔列表。

In [31]:
arr = [1,2,3,4,5]
s = ','.join(str(n) for n in arr)
s

'1,2,3,4,5'

### 练习函数调用。

In [32]:
import time
def hello_world():
    print('hello world')
def three_hellos():
    for i in range(0, 3):
        hello_world()
        time.sleep(1)
if __name__ == '__main__':
    three_hellos()

hello world
hello world
hello world


### 文本颜色设置。

In [33]:
class bcolors:
    HEADER = '\033[95m'
    OKBLUE = '\033[94m'
    OKGREEN = '\033[92m'
    WARNING = '\033[93m'
    FAIL = '\033[91m'
    ENDC = '\033[0m'
    BOLD = '\033[1m'
    UNDERLINE = '\033[4m'
print(bcolors.FAIL + "警告的颜色字体?" + bcolors.ENDC)

警告的颜色字体?


### 求100之内的素数。

In [34]:
def findSingleNumber(n):
    arr = []
    for i in range(2, n+1):
        flag = True
        for j in range(2, int(i/2)+1):
            if i%j ==0:
                flag = False
                break
        if flag:
            arr.append(i)
    print(arr)
findSingleNumber(100)

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]


### 对10个数进行排序

In [35]:
def sortNumbers(arr):
    for i in range(0, len(arr)):
        for j in range(i, len(arr)):
            if arr[i] > arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
    print(arr)
sortNumbers([2,3,5,1,10, 10,123,4])

[1, 2, 3, 4, 5, 10, 10, 123]


### 求一个3*3矩阵主对角线元素之和。

In [36]:
arr = [[1,2,3],[4,5,6],[7,8,9]]
def calculateLineSum(arr):
    i= 0
    sumRe = 0
    for k in range(0, 3):
        sumRe+=arr[i][i]
        i+=1
    print(sumRe)
calculateLineSum(arr)

15


### 有一个已经排好序的数组。现输入一个数，要求按原来的规律将它插入数组中。

In [37]:
def insertNumber(arr, n):
    #rr.append(n)
    indx = -1
    result = []
    for i in range(0, len(arr)-1):
        if (arr[i]<=n and arr[i+1]>=n) or (arr[i]>=n and arr[i+1]<=n):
            indx = i
            break
    if indx !=-1:
        result= arr[:indx+1]+[n]+arr[indx+1:]
    elif (n>arr[0] and arr[0]<=arr[len(arr)-1]) or (n<arr[0] and arr[0]>=arr[len(arr)-1]):
        result = arr+[n]
    elif (n<arr[0]and arr[0]<=arr[len(arr)-1]) or (n>arr[0] and arr[0]>=arr[len(arr)-1]):
        result = [n]+arr
    else:
        print('data error')
    print(result)
insertNumber([0,0,2,3], -1)

[-1, 0, 0, 2, 3]


### 将一个数组逆序输出。

In [38]:
def reverPrintArray(arr):
    for i in arr[::-1]:
        print(i)
reverPrintArray([9,6,5,4,3])

3
4
5
6
9


### 两个 3 行 3 列的矩阵，实现其对应位置的数据相加，并返回一个新矩阵：

In [39]:
X = [[12,7,3],
    [4 ,5,6],
    [7 ,8,9]]

Y = [[5,8,1],
    [6,7,3],
    [4,5,10]]
def addMatrix(x, y):
    arr = []
    for i in range(0, len(x)):
        brr =[]
        for j in range(0, len(x[i])):
            brr.append(x[i][j]+y[i][j])
        arr.append(brr)
    print(arr)
addMatrix(X, Y)

[[17, 15, 4], [10, 12, 9], [11, 13, 19]]


### 统计 1 到 100 之和。

In [47]:
def countSum(n):
    return sum(range(1, n+1))
countSum(100)

5050

利用for循环控制100-999个数，每个数分解出个位，十位，百位。。### 

In [50]:
if __name__  == '__main__':
    from tkinter import *
    canvas = Canvas(width = 300,height = 300,bg = 'green')
    canvas.pack(expand = YES,fill = BOTH)
    x0 = 150
    y0 = 100
    canvas.create_oval(x0 - 10,y0 - 10,x0 + 10,y0 + 10)
    canvas.create_oval(x0 - 20,y0 - 20,x0 + 20,y0 + 20)
    canvas.create_oval(x0 - 50,y0 - 50,x0 + 50,y0 + 50)
    import math
    B = 0.809
    for i in range(16):
        a = 2 * math.pi / 16 * i
        x = math.ceil(x0 + 48 * math.cos(a))
        y = math.ceil(y0 + 48 * math.sin(a) * B)
        canvas.create_line(x0,y0,x,y,fill = 'red')
    canvas.create_oval(x0 - 60,y0 - 60,x0 + 60,y0 + 60)
    

    for k in range(501):
        for i in range(17):
            a = (2 * math.pi / 16) * i + (2 * math.pi / 180) * k
            x = math.ceil(x0 + 48 * math.cos(a))
            y = math.ceil(y0 + 48 + math.sin(a) * B)
            canvas.create_line(x0,y0,x,y,fill = 'red')
        for j in range(51):
            a = (2 * math.pi / 16) * i + (2* math.pi / 180) * k - 1
            x = math.ceil(x0 + 48 * math.cos(a))
            y = math.ceil(y0 + 48 * math.sin(a) * B)
            canvas.create_line(x0,y0,x,y,fill = 'red')
    mainloop()

### 打印字符串

In [51]:
sStr1='abcdefg'
sStr2='cde'
print(sStr1.find(sStr2))

2


### 画椭圆

In [55]:
if __name__ == '__main__':
    from tkinter import *
    x  = 360
    y = 160
    top = y - 30
    bottom = y - 30
    
    canvas = Canvas(width = 400, height = 600, bg = 'white')
    for i in range(20):
        canvas.create_oval(250 - top, 250 - bottom, 250 + top, 250 + bottom)
        top -= 5
        bottom += 5
    canvas.pack()
    mainloop()

### 利用ellipse 和 rectangle 画图。。

In [56]:
#!/usr/bin/python
# -*- coding: UTF-8 -*-

if __name__ == '__main__':
    from tkinter import *
    canvas = Canvas(width = 400,height = 600,bg = 'white')
    left = 20
    right = 50
    top = 50
    num = 15
    for i in range(num):
        canvas.create_oval(250 - right,250 - left,250 + right,250 + left)
        canvas.create_oval(250 - 20,250 - top,250 + 20,250 + top)
        canvas.create_rectangle(20 - 2 * i,20 - 2 * i,10 * (i + 2),10 * ( i + 2))
        right += 5
        left += 5
        top += 10

    canvas.pack()
    mainloop()